src/main/java/com/conveyal/analysis/persistence/AnalysisCollection.java

```diff
@@ -73,6 +73,14 @@ public T create(T newModel, String accessGroup, String creatorEmail) {
         return newModel;
     }
 
+    /**
+     * Note that if the supplied model has _id = null, the Mongo insertOne method will overwrite it with a new
+     * ObjectId(). We consider it good practice to set the _id for any model object ourselves, avoiding this behavior.
```

Comment:
Should we add an assertion here to require ourselves to follow this convention? _id must already exist before insert?

Reply:
It looks like we could remove the OBJECT_ID_GENERATORS convention to force explicit ID creation. https://mongodb.github.io/mongo-java-driver/3.11/bson/pojos/#conventions